### PR TITLE
Merge xr hyper

### PR DESCRIPTION
**Pull request by u-fischer**
This merges the code from xr-hyper into xr. With it xr will support links and url, it will also support properties (from #1180). xr-hyper will be obsolete.

# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [x] Rollback provided (if necessary)? The previous version can be loaded with [=v5]
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: required/tools/xr.dtx [hunk 1]
```diff
@@ -21,14 +21,49 @@
...
+%<package>\DeclareRelease{v5}{2023-07-04}{xr-2023-07-04.sty}
+%<package>\DeclareCurrentRelease{}{2024-04-10}
 %<package>\ProvidesPackage{xr}
-%<package>         [2023-07-04 v5.06 eXternal References (DPC)]
+%<package>         [2024-04-10 v6.00 eXternal References (DPC)]
 %
 %<*driver>
-\documentclass{ltxdoc}
+\documentclass{l3doc}
+% currently missing in l3doc, copied from array.dtx
```
Comment (FrankMittelbach) [inline required/tools/xr.dtx:34]: Iif there isn't already we should open an issue to get this added and then removed in various doc files.

File: base/doc/ltnews39.tex [hunk 2]
```diff
@@ -659,7 +659,18 @@ \subsection{\pkg{showkeys}: Allow \cs{newline} in \pkg{amsthm} to work}
...
 \githubissue{1123}
 
-
+\subsection{\pkg{xr}: Support links and properties}
+%
+The \pkg{xr} package implements a system for eXternal References. 
+The \pkg{xr-hyper} package (in the \pkg{hyperref} bundle) 
+extended this to also support links to external documents. With
+the extension of the \cs{label} command last year that unified 
+the label syntax of \LaTeX{} and \pkg{hyperref} is now possible
```
Comment (mrpiggi) [inline base/doc/ltnews39.tex:668]: Should read "it is now possible"

**Review by FrankMittelbach — COMMENTED** [review]
can we have a test against using v5 and also using xr-hyper?

File: required/tools/xr.dtx (hunk 1 above)
Comment (u-fischer) [reply required/tools/xr.dtx:34]: there is an issue: https://github.com/latex3/latex3/issues/1359

**Comment by u-fischer** [comment]
> can we have a test against using v5 and also using xr-hyper?

I added two tests.

File: required/tools/xr.dtx (hunk 1 above)
Comment (FrankMittelbach) [reply required/tools/xr.dtx:34]: tough, even by me .. I  thought I did, but forgot that would have been in latex3 ...

**Review by FrankMittelbach — APPROVED** [review]

**Pull request merged**